library/core/src/main/java/com/google/android/exoplayer2/extractor/mp4/FragmentedMp4Extractor.java

```diff
@@ -1276,10 +1277,17 @@ private boolean readSample(ExtractorInput input) throws IOException, Interrupted
         }
       }
     } else {
+      int sampleHeaderSize = 0;
```

Comment:
The code added here needs to go in the `if (parserState == STATE_READING_SAMPLE_START)` block above because the call to `output.sampleData(input, ...)` below may throw if there's a problem reading from the source, at which point we retry reading (and as things stand we'd output the header again). Ditto for `Mp4Extractor`.

Reply:
I see. I'll update the code.